test/jasmine/tests/calcdata_test.js

```diff
@@ -28,4 +29,215 @@ describe('calculated data and points', function() {
             expect(gd.calcdata[0][3]).toEqual({ x: false, y: false});
         });
     });
+
+    xdescribe('category ordering', function() {
```

Comment:
@monfera these tests are great. The test :tiger2: thanks you! 

Could you add a couple more cases where multiple traces with [shared](http://codepen.io/etpinard/pen/JXOMVZ) and/or [mutually exclusive](http://codepen.io/etpinard/pen/ZQxxWb) categories?